src/topology/separation.lean

```diff
@@ -654,6 +654,28 @@ begin
   exact (set.to_finite _).is_closed
 end
 
+lemma preconnected_space.trivial_of_discrete [preconnected_space α] [discrete_topology α] :
+  subsingleton α :=
+begin
+  rw ←not_nontrivial_iff_subsingleton,
+  rintro ⟨x, y, hxy⟩,
+  obtain h | h := is_clopen_iff.mp (is_clopen_discrete ({y} : set α)),
+  { exact (singleton_nonempty y).ne_empty h },
+  { rw [ne.def, ←mem_singleton_iff, h] at hxy,
+    exact hxy (mem_univ x) }
+end
+
+lemma is_connected.infinite_of_nontrivial [t1_space α] {s : set α} (h : is_connected s)
```

Comment:
can you weaken the assumption to `is_preconnected s`, as nonemptyness is already contained in the other assumption?

Reply:
sure; I left the instance still on `connected_space`, because I'd feel weird writing `[no_zero_divisors] [nontrivial]` instead of `[is_domain] [nontrivial]`, so I applied the same logic. I don't mind either way though.